cloud-openstack-server/src/main/java/jetbrains/buildServer/clouds/openstack/OpenstackApi.java

```diff
@@ -14,72 +16,119 @@
 import org.jclouds.openstack.nova.v2_0.domain.Image;
 import org.jclouds.openstack.nova.v2_0.features.ServerApi;
 
-import java.util.List;
-import java.util.Properties;
+import com.google.common.collect.ImmutableSet;
 
+import jetbrains.buildServer.util.StringUtil;
 
 public class OpenstackApi {
 
-    protected final String endpointUrl;
-    protected final String identity;
-    protected final String password;
-
     private final String zone;
 
-    private NeutronApi neutronApi;
-    private NovaApi novaApi;
+    private final NeutronApi neutronApi;
+    private final NovaApi novaApi;
 
     public OpenstackApi(String endpointUrl, String identity, String password, String zone) {
 
-        this.endpointUrl = endpointUrl;
-        this.identity = identity;
-        this.password = password;
         this.zone = zone;
 
         final Properties overrides = new Properties();
-        overrides.setProperty(KeystoneProperties.CREDENTIAL_TYPE, CredentialTypes.PASSWORD_CREDENTIALS);
-        overrides.setProperty(Constants.PROPERTY_API_VERSION, "2");
+        final String keyStoneVersion = getKeystoneVersion(endpointUrl);
+        final String[] identityArray = getIdentityArray(identity);
+        overrides.setProperty(KeystoneProperties.KEYSTONE_VERSION, keyStoneVersion);
         overrides.setProperty(LocationConstants.PROPERTY_ZONES, zone);
-
-        novaApi = ContextBuilder.newBuilder(new NovaApiMetadata())
-                .endpoint(endpointUrl)
-                .credentials(identity, password)
-                .overrides(overrides)
-                .buildApi(NovaApi.class);
+        overrides.setProperty(KeystoneProperties.SCOPE, "project:" + identityArray[1]);
 
         neutronApi = ContextBuilder.newBuilder(new NeutronApiMetadata())
-                .credentials(identity, password)
-                .endpoint(endpointUrl)
-                .overrides(overrides)
-                .buildApi(NeutronApi.class);
+                .credentials(getCredentialsFromIdentity(identityArray, keyStoneVersion), password).endpoint(endpointUrl).overrides(overrides)
+                .modules(ImmutableSet.of(new SLF4JLoggingModule())).buildApi(NeutronApi.class);
```

Comment:
The log implementation should be reviewed ... this change occurs an error when plugin is really used into TeamCity:
```
[...]
at jetbrains.buildServer.clouds.openstack.OpenstackApi.<init>(OpenstackApi.java:45)
[...]
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.jclouds.logging.slf4j.SLF4JLogger
[...]
```